% why we want to predict/understand TR
Understanding how \gls{mrna} concentration in eukaryotes cells (\acrlong{tr}) is regulated is very important to understand gene expression. These concentrations can vary depending on the synthesis and/or degradation rates and the rapidity of the transition to a new concentration depends on the regulation of \gls{mrna} stability \cite{PEREZORTIN2007250}. \fxnote{Fix/rewrite this after finishing the results and bio back.}

% why we use images to predict transcription rate?
However, the protein or \gls{mrna} amount inside a cell, may not be enough to fully describe cellular function. Accordingly to Buxbaum et al. \cite{Buxbaum_2014} and Korolchuk et al. \cite{Korolchuk2011}, cellular function can heavily depends on the specific intracellular location and interaction with other molecules and intracellular structures. Therefore, cellular expression is determined by the functional state, abundance, morphology, and turnover of its intracellular organelles and cytoskeletal structures. This means that, models that predict \gls{tr} by means of single cell images, may be able to leverage scientific discoveries in biomedicine.

% encourage model to rely on spatial information
Mention here that we ofuscate the color and break the color channel correlation in order to force the model to predict tr using the spatial information (cell nucleus structure)
