%\glsresetall
% define where the images are
\graphicspath{{./Sections/Basics/Resources/}}

Transcription is the process in which the instructions encoded in a gene are copied from the DNA inside the nucleus, to produce a RNA transcript called \acrlong{mrna}. The Transcription process has two broad 2 main steps; 1) The process in which the gene is copied from the DNA into a pre-processed version of the \gls{mrna} called \gls{pmrna}; 2) The RNA Splicing, which is the process where the \gls{pmrna} is transformed into a mature \gls{mrna} strand. These two processes happen inside the cell nucleus.

\subsubsection{Step 1, Pre-messenger RNA synthesis}
The \gls{pmrna} creation has 3 main processes\cite{MolecularWatson} and is illustrated in figure \ref{fig:BB:premrna_synth}:

\begin{enumerate}
  \item \textbf{Initiation}. This step initiate the transcription process. It happen when an \textit{enzyme}\footnote{An enzyme is a proteins that act as biological catalysts to accelerate chemical reactions.} of RNA polymerase binds to the a region of the target gene called \textit{the promoter}. This indicate the DNA to unwind, so the RNA polymerase can read the DNA bases in one of its strands and create a molecule of \gls{pmrna}.
  \item \textbf{Elongation}. Elongation is the process in which \textit{nucleotides}\footnote{Nucleotides are the building block of nucleic acids. A nucleotide consists of a sugar molecule bound to a phosphate group and a nitrogen-containing base. RNA and DNA are polymers made of long chains of nucleotides.} are added to the \gls{pmrna} strand. The RNA polymerase enzyme read the unwound DNA strand and synthesize the \gls{pmrna} molecule.
  \item \textbf{Termination}. Termination ends the \gls{pmrna} synthesis. It happens when the RNA polymerase enzyme identifies a termination sequence in the gene, and detaches from the unwound DNA.
\end{enumerate}

\begin{figure}[htb]
  \centering
  \includegraphics[width=0.5\linewidth]{trans_steps.png}
  \caption{The three main steps of the \gls{pmrna} synthesis: initiation, elongation, and termination \cite{transcription_steps}.}
  \label{fig:BB:premrna_synth}
\end{figure}

\subsubsection{Step 2, Pre-messenger RNA splicing}
In this process the \gls{pmrna} is transformed into a mature \gls{mrna} strand by removing non-relevant (non-coding) sections of it.
During Splicing, \textit{introns}\footnote{Introns are nucleotide sequences within a gene that are non-coding regions of an RNA transcript, and that are removed by the splicing process before translation.} are removed and \textit{exons}\footnote{Exons are coding sections of an RNA transcript that can be translated into proteins.} are joined together \cite{Biochemistry}. Besides this, a cap and a tail are added to the spliced \gls{pmrna} strand to turn it into a mature \gls{mrna}. Figure \ref{fig:BB:splicing} illustrates the \gls{pmrna} splicing process.

\begin{figure}[htb]
  \centering
  \includegraphics[width=0.7\linewidth]{splicing_process.png}
  \caption{Pre-messenger RNA splicing process. A \gls{pmrna} strand (top) is turned into a mature \gls{mrna} strand (bottom) \cite{wiki:Primary_transcript}.}
  \label{fig:BB:splicing}
\end{figure}

\subsubsection{Transcription Rate}

Here I should state what is \gls{tr}. In this part I will not mention that we use the average of the channel \texttt{00\_EU}, that will be stated in the methodology part. However, here I should state a definition of \gls{tr} that agrees with what we have in the dataset. Here are some ideas of what \gls{tr} is.
Since we have images of cell nucleus, this means that we have both \gls{pmrna} and mature \gls{mrna} molecules\footnote{Transcription rate encompasses two related, yet different, concepts: the nascent transcription rate, which measures the in situ mRNA production by RNA polymerase, and the rate of synthesis of mature mRNA, which measures the contribution of transcription to the mRNA concentration \cite{what_is_tr}.}. So, what is \texttt{00\_EU} measuring? both molecules or only one?
\fxnote{Discuss this with Hannah!}
As far as I understood, \texttt{00\_EU} captures the \gls{mrna} molecules (pre or mature?) that were produced in the last 30 minutes. Therefore, for us \gls{tr} would be the average concentration of \gls{mrna} per unit of time (in this case 30 mins). However if we measure the concentration of pre or mature \gls{mrna} in the nucleus, then in the \gls{tr} definition we should consider either (or both) the rate that pre turns into mature (but if we measure both then we mesure the total and this is not necessary) or the rate at which mature \gls{mrna} escapes the nucleus.

Therefore, accordingly to \cite{what_is_tr} we can define \gls{tr} as the number of \gls{mrna} molecules being synthesized per unit time minus a degradation factor
\begin{equation}
  \frac{d[mRNA]}{dt} = SR - k_d [mRNA]
\end{equation}
\noindent where $[mRNA]$ is the mature \gls{mrna} concentration in the cell nucleus, $RN$ the \gls{mrna} synthesis rate and $k_d$ the degradation rate (or in our case, the rate at which mature \gls{mrna} escapes the nucleus).
