%% Magic command to compile root document
% !TEX root = ../../thesis.tex

\glsresetall
% define where the images are
\graphicspath{{./Sections/Basics/Resources/}}

Transcription is the process in which the instructions encoded in a gene are copied from the DNA inside the nucleus, to produce a RNA transcript called \acrlong{mrna}. The Transcription process has two broad 2 main steps; 1) The process in which the gene is copied from the DNA into a pre-processed version of the \gls{mrna} called \gls{pmrna}; 2) The RNA Splicing, which is the process where the \gls{pmrna} is transformed into a mature \gls{mrna} strand. These two processes happen inside the cell nucleus.

\subsubsection{Step 1, Pre-messenger RNA synthesis}
The \gls{pmrna} creation has 3 main processes\cite{MolecularWatson} and is illustrated in figure \ref{fig:BB:premrna_synth}:

\begin{enumerate}
  \item \textbf{Initiation}. This step initiate the transcription process. It happen when an \textit{enzyme}\footnote{An enzyme is a proteins that act as biological catalysts to accelerate chemical reactions.} of RNA polymerase binds to a region of the target gene called \textit{the promoter}. This indicate the DNA to unwind, so the RNA polymerase can read the DNA bases in one of its strands and create a molecule of \gls{pmrna}.
  \item \textbf{Elongation}. Elongation is the process in which \textit{nucleotides}\footnote{Nucleotides are the building block of nucleic acids. A nucleotide consists of a sugar molecule bound to a phosphate group and a nitrogen-containing base. RNA and DNA are polymers made of long chains of nucleotides.} are added to the \gls{pmrna} strand. The RNA polymerase enzyme read the unwound DNA strand and synthesize the \gls{pmrna} molecule.
  \item \textbf{Termination}. Termination ends the \gls{pmrna} synthesis. It happens when the RNA polymerase enzyme identifies a termination sequence in the gene, and detaches from the unwound DNA.
\end{enumerate}

\begin{figure}[htb]
  \centering
  \includegraphics[width=0.5\linewidth]{trans_steps.png}
  \caption{The three main steps of the \gls{pmrna} synthesis: initiation, elongation, and termination. Image source \cite{transcription_steps}.}
  \label{fig:BB:premrna_synth}
\end{figure}

\subsubsection{Step 2, Pre-messenger RNA splicing}
In this process the \gls{pmrna} is transformed into a mature \gls{mrna} strand by removing non-relevant (non-coding) sections of it.
During Splicing, \textit{introns}\footnote{Introns are nucleotide sequences within a gene that are non-coding regions of an RNA transcript, and that are removed by the splicing process before translation.} are removed and \textit{exons}\footnote{Exons are coding sections of an RNA transcript that can be translated into proteins.} are joined together \cite{Biochemistry}. Besides this, a cap and a tail are added to the spliced \gls{pmrna} strand to turn it into a mature \gls{mrna}. The splicing process takes place within subnuclear structures called \gls{ns} (also known as \hl{Splicing Speckles}) \cite{spector2011nuclear}.
Figure \ref{fig:BB:splicing} illustrates the \gls{pmrna} splicing process.

\begin{figure}[htb]
  \centering
  \includegraphics[width=0.7\linewidth]{splicing_process.png}
  \caption{Pre-messenger RNA splicing process. A \gls{pmrna} strand (top) is turned into a mature \gls{mrna} strand (bottom). Image source \cite{wiki:Primary_transcript}.}
  \label{fig:BB:splicing}
\end{figure}

\subsubsection{Transcription Rate}

\gls{tr} contemplates the two steps we already explained; 1) the nascent transcription rate, which measures the in situ \gls{mrna} produced by the RNA polymerase enzyme (\gls{pmrna}), and 2) the rate of synthesis of mature \gls{mrna} (\gls{pmrna} splicing), which measures the contribution of transcription to the \gls{mrna} concentration \cite{what_is_tr}.

Accordingly to Pérez-Ortín et al. \cite{what_is_tr}, we can define the change in the mature \gls{mrna} concentration ([mRNA]), as the number of mature \gls{mrna} molecules being synthesized per unit of time minus a degradation factor
\begin{equation}
  \frac{d[mRNA]}{dt} = SR - k_d [mRNA]
\end{equation}
\noindent where $[mRNA]$ is the mature \gls{mrna} concentration in the cell nucleus, $SR$ the mature \gls{mrna} synthesis rate and $k_d$ the degradation rate.

However, the objective of this work is not to model the change in \gls{tr} across time. Instead, we are only interested in estimating the \gls{tr} of a cell, given a snapshot of it at a specific moment of time. For this reason, in this work we will understand \gls{tr} as the amount of nascent \gls{mrna} in a given period of time \cite{wansink1993fluorescent}.
